pkg/servicemesh/apis/servicemesh/v1alpha1/servicemeshextension_types.go

```diff
@@ -15,16 +15,19 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ServiceMeshExtensionSpec defines the desired state of ServiceMeshExtension
 type ServiceMeshExtensionSpec struct {
-	Image            string           `json:"image,omitempty"`
-	WorkloadSelector WorkloadSelector `json:"workloadSelector,omitempty"`
-	Phase            *FilterPhase     `json:"phase"`
-	Priority         *int             `json:"priority,omitempty"`
-	Config           string           `json:"config,omitempty"`
+	Image            string                        `json:"image,omitempty"`
+	ImagePullPolicy  corev1.PullPolicy             `json:"imagePullPolicy,omitempty"`
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
+	WorkloadSelector WorkloadSelector              `json:"workloadSelector,omitempty"`
+	Phase            *FilterPhase                  `json:"phase"`
+	Priority         *int                          `json:"priority,omitempty"`
+	Config           string                        `json:"config,omitempty"`
```

Comment:
this is a `struct{}` in the new API -- it'll have to be converted to json by istiod

Reply:
In your API PR this is a `google.protobuf.Struct` in .proto (which maps to a proto struct in go). What do you have in mind for Maistra? Keep it string? map[string]interface{}? It's currently passed straight to envoy here: https://github.com/maistra/istio/blob/ea314b69a97fde3102b758ff5dea592cf34c5713/pkg/servicemesh/extension/apply_extension.go#L169